x-pack/filebeat/input/s3/input.go

```diff
@@ -248,16 +248,16 @@ func (p *s3Input) processMessage(svcS3 s3iface.ClientAPI, message sqs.Message, w
 
 	s3Infos, err := handleSQSMessage(message)
 	if err != nil {
-		p.logger.Error(errors.Wrap(err, "handleMessage failed"))
+		p.logger.Error(errors.Wrap(err, "handleSQSMessage failed"))
 		return
 	}
 
 	// read from s3 object and create event for each log line
 	err = p.handleS3Objects(svcS3, s3Infos, errC)
 	if err != nil {
 		err = errors.Wrap(err, "handleS3Objects failed")
-		errC <- err
 		p.logger.Error(err)
+		errC <- err
```

Comment:
Switching the order here so the error message get logged first and then send to error channel. Otherwise the error log never get published.